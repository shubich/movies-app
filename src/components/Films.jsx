import React from 'react';
import {Film} from './Film'

const movies = [
    {
        title: 'KILL BILL: VOL. 2',
        year: '2004',
        genre: 'Action & Adventure',
        img: 'https://images-na.ssl-images-amazon.com/images/M/MV5BNmFiYmJmN2QtNWQwMi00MzliLThiOWMtZjQxNGRhZTQ1MjgyXkEyXkFqcGdeQXVyNzQ1ODk3MTQ@._V1_.jpg'
    },
    {
        title: 'KILL BILL: VOL. 1',
        year: '2003',
        genre: 'Action & Adventure',
        img: 'https://images-na.ssl-images-amazon.com/images/M/MV5BYTczMGFiOWItMjA3Mi00YTU5LWIwMDgtYTEzNjRkNDkwMTE2XkEyXkFqcGdeQXVyNzQ1ODk3MTQ@._V1_.jpg'
    },
    {
        title: 'PULP FICTION',
        year: '1994',
        genre: 'Oscar-winning Movies',
        img: 'https://images-na.ssl-images-amazon.com/images/M/MV5BMTkxMTA5OTAzMl5BMl5BanBnXkFtZTgwNjA5MDc3NjE@._V1_SY1000_CR0,0,673,1000_AL_.jpg'
    },
    {
        title: 'JACKIE BROWN',
        year: '1997',
        genre: 'Dramas',
        img: 'https://images-na.ssl-images-amazon.com/images/M/MV5BNmY5ODRmYTItNWU0Ni00MWE3LTgyYzUtYjZlN2Q5YTcyM2NmXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SY1000_CR0,0,683,1000_AL_.jpg'
    },
    {
        title: 'JACKIE BROWN',
        year: '1997',
        genre: 'Dramas',
        img: 'https://images-na.ssl-images-amazon.com/images/M/MV5BNmY5ODRmYTItNWU0Ni00MWE3LTgyYzUtYjZlN2Q5YTcyM2NmXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SY1000_CR0,0,683,1000_AL_.jpg'
    }
];

export const Films = () => (
    <div id='films'>
        {
            movies.map((movie, i) => (
                <Film key={i} {...movie}/>
            ))
        }
    </div>
);