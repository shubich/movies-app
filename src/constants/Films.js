// export const GET_FILMS_BY_TITLE = 'GET_FILMS_BY_TITLE';
// export const GET_FILMS_BY_DIRECTOR = 'GET_FILMS_BY_DIRECTOR';
// export const SORT_FILMS_BY_DATE = 'SORT_FILMS_BY_DATE';
// export const SORT_FILMS_BY_RATING = 'SORT_FILMS_BY_RATING';
export const GET_FILMS = 'GET_FILMS';
export const SORT_FILMS = 'SORT_FILMS';