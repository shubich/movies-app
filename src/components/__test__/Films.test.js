import React from 'react';
import { Films } from '../Films';
import { shallow } from 'enzyme';

let props;

describe('Films', () => {
    beforeEach(() => {
        props = {
            results: [
                {
                    id: 1,
                    title: 'A'
                },
                {
                    id: 2,
                    title: 'B'
                }
            ],
            fetching: false,
            error: '',
            onFilmClick: jest.fn
        };
    });

    it('renders correctly with results', () => {
        const wrapper = shallow(
            <Films {...props} />
        );
        expect(wrapper).toMatchSnapshot();
    });


    it('renders correctly with fetching', () => {
        const wrapper = shallow(
            <Films {...props} fetching={true} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly without results', () => {
        const wrapper = shallow(
            <Films {...props} results={[]} />
        );
        expect(wrapper).toMatchSnapshot();
    });
})