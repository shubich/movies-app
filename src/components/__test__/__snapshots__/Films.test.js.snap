// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films renders correctly with fetching 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Films
    error=""
    fetching={true}
    onFilmClick={[Function]}
    results={
      Array [
        Object {
          "id": 1,
          "title": "A",
        },
        Object {
          "id": 2,
          "title": "B",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <p>
          Loading...
        </p>
      </div>,
      "id": "films",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Loading...
        </p>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <p>
            Loading...
          </p>
        </div>,
        "id": "films",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Loading...
          </p>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Films renders correctly with results 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Films
    error=""
    fetching={false}
    onFilmClick={[Function]}
    results={
      Array [
        Object {
          "id": 1,
          "title": "A",
        },
        Object {
          "id": 2,
          "title": "B",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <Film
          film={
            Object {
              "id": 1,
              "title": "A",
            }
          }
          onClick={[Function]}
        />
        <Film
          film={
            Object {
              "id": 2,
              "title": "B",
            }
          }
          onClick={[Function]}
        />
      </div>,
      "id": "films",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Film
            film={
              Object {
                "id": 1,
                "title": "A",
              }
            }
            onClick={[Function]}
          />,
          <Film
            film={
              Object {
                "id": 2,
                "title": "B",
              }
            }
            onClick={[Function]}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "film": Object {
              "id": 1,
              "title": "A",
            },
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "film": Object {
              "id": 2,
              "title": "B",
            },
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <Film
            film={
              Object {
                "id": 1,
                "title": "A",
              }
            }
            onClick={[Function]}
          />
          <Film
            film={
              Object {
                "id": 2,
                "title": "B",
              }
            }
            onClick={[Function]}
          />
        </div>,
        "id": "films",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Film
              film={
                Object {
                  "id": 1,
                  "title": "A",
                }
              }
              onClick={[Function]}
            />,
            <Film
              film={
                Object {
                  "id": 2,
                  "title": "B",
                }
              }
              onClick={[Function]}
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "film": Object {
                "id": 1,
                "title": "A",
              },
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "film": Object {
                "id": 2,
                "title": "B",
              },
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Films renders correctly without results 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Films
    error=""
    fetching={false}
    onFilmClick={[Function]}
    results={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
